validateWorkerVersions.ps1

```diff
@@ -0,0 +1,87 @@
+<#
+    .SYNOPSIS
+        Used to validate and/or update worker package versions
+    .EXAMPLE
+        ./validateWorkerVersions.ps1
+
+        Validates the workers match the existing host version and throws an error if they don't
+    .EXAMPLE
+        ./validateWorkerVersions.ps1 -Update -HostVersion 4.0.0
+
+        Updates the host reference to 4.0.0 and the workers to their matching versions
+#>
+param (
+    [Switch]$Update,
+    
+    # An explicit host version, otherwise the host version from Azure.Functions.Cli.csproj will be used
+    [string]$hostVersion
+)
+
+# the xml will fail to parse if the data is encoded with a bom character
+function removeBomIfExists([string]$data)
+{
+    if ($data.StartsWith(0xFEFF)) {
+        $data = $data.substring(1)
+    }
+    return $data
+}
+
+$cliCsprojPath = "./src/Azure.Functions.Cli/Azure.Functions.Cli.csproj"
+$cliCsprojContent = removeBomIfExists(Get-Content $cliCsprojPath)
+$cliCsprojXml = [xml]$cliCsprojContent
+
+function getPackageVersion([string]$packageName, [string]$csprojContent)
+{
+    $version = (Select-Xml -Content $csprojContent -XPath "/Project//PackageReference[@Include='$packageName']/@Version").ToString()
+    if (-Not $version) {
+        throw "Failed to find version for package $packageName"
+    }
+    return $version
+}
+
+function setCliPackageVersion([string]$packageName, [string]$version)
+{
+    $node = $cliCsprojXml.SelectSingleNode("/Project//PackageReference[@Include='$packageName']")
+    $node.Version = $version

```

Comment:
Having a log entry to says what `version` the worker is being updated to would be helpful to debug without having to rerun the script.